koitoror/SendIT
# Ft parcels endpoints 161738123

#### What does this PR do?
Creates all the required parcel delivery order API endpoints to : 
- Fetch all parcel delivery orders - for all users [by admin]
- Create a parcel delivery order [by user]
- Fetch a single parcel delivery order [by user]
- Delete a single parcel delivery order [by admin]
- Modify present location and status of a parcel delivery order [by admin]
- Modify the status, to cancel, a delivery order [by user]
- Fetch all parcel delivery orders - for single user [by user]


#### Description of Task to be completed?
Have the `parcels.py` working


#### How should this be manually tested?
For the Tests to work you need do the following steps

Clone the repository into your local environment

```
$ git clone https://github.com/koitoror/SendIT.git
```

Switch to the PARCELS branch
```
$ git checkout ft-parcels-endpoints-161738123
```

Navigate to api folder.
```
cd SendIT
```

Create a virtual enviroment.
Use [virtualenv](http://www.pythonforbeginners.com/basics/how-to-use-python-virtualenv) for an isolated working environment.

```
virtualenv venv --python=python3
```

Install the packages.
```
pip3 install -r requirements.txt
```

Set environment variables for 

> `SECRET_KEY` is your secret key

> `FLASK_CONFIG` is the enviroment you are running on. Should be either `Production`, `Development` or `Testing`. NOTE: its case sensitive

To get the app running...

```bash
$ python run.py run
```

Open root path in your browser to test the endpoints. 
You can also use Postman or any other agent to test the endpoints

On Postman enter the relevant URL from below list which is

**EndPoint** | **Functionality**
--- | ---
GET  `/api/v1/parcels` | Fetch all parcel delivery orders
POST  `/api/v1/parcels` | Create a parcel delivery order
GET  `/api/v1/parcels/<parcelId>` | Fetch a single parcel delivery order 
PUT  `/api/v1/parcels/<parcelId>` | Update a parcel delivery order by admin
DELETE  `/api/v1/parcels/<parcelId>` | Delete a parcel delivery order by admin
PUT  `/api/v1/parcels/<parcelId>/cancel` | Cancel a parcel delivery order by user
GET  `/api/v1/users/<userId>/parcels` | Fetch all parcel delivery orders by user


#### Any background context you want to provide?
- Loaded the app runners, blueprints,  tests integrations and configurations.
- Did chores of updating the Files, Directory Restructure and Code refactors

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2216466
#161738123

#### Screenshots (if appropriate)
